### PR TITLE
[Geomagic] Update GeomagicDriver.cpp to fire error when hd.h is not found

### PR DESCRIPTION
**Pull request by epernod**
quick PR from the pool regarding discussion #4561

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Review by bakpaul — APPROVED** [review]

File: applications/plugins/Geomagic/src/Geomagic/GeomagicDriver.cpp [hunk 1]
```diff
@@ -346,6 +346,7 @@ void GeomagicDriver::initDevice()
     updatePosition();
     sofa::core::objectmodel::BaseObject::d_componentState.setValue(sofa::core::objectmodel::ComponentState::Valid);
 #else
+    msg_error() << "GeomagicDriver initialisation failed because OpenHaptics (HD/hd.h) has not been found. Please install openHaptics and specify OpenHaptics_DIR at CMake configure stage.";
```
Comment (alxbilger) [inline applications/plugins/Geomagic/src/Geomagic/GeomagicDriver.cpp]:
```suggestion
    msg_error() << "GeomagicDriver initialization failed because the Geomagic plugin was built without OpenHaptics (HD/hd.h). If you build the plugin yourself, install OpenHaptics and specify OpenHaptics_DIR at CMake configure stage.";
```

As this message is intended for users, I would rephrase like that

**Review by alxbilger — CHANGES_REQUESTED** [review]
Can you remind me why we compile this code even if OpenHaptics is not found?

File: applications/plugins/Geomagic/src/Geomagic/GeomagicDriver.cpp (hunk 1 above)
Comment (epernod) [reply applications/plugins/Geomagic/src/Geomagic/GeomagicDriver.cpp]: It was to compile the plugin on the CI, where openHaptics can't be installed,  to catch any error due to change in SOFA base of code.
And I wanted to create an emulator AT one moment... Still in m'y todo list...
This was the solution 3 years ago. Maybe WE can install HD now on the CI?

Comment (epernod) [reply applications/plugins/Geomagic/src/Geomagic/GeomagicDriver.cpp]: Ok for the change I let you validate the final version during the meeting.
